ocs_ci/ocs/ocs_upgrade.py

```diff
@@ -0,0 +1,363 @@
+from copy import deepcopy
+from pkg_resources import parse_version
+from tempfile import NamedTemporaryFile
+
+from ocs_ci.deployment.deployment import create_catalog_source
+from ocs_ci.framework import config
+from ocs_ci.ocs import constants
+from ocs_ci.ocs.defaults import OCS_OPERATOR_NAME
+from ocs_ci.ocs.node import get_typed_nodes
+from ocs_ci.ocs.ocp import get_images, OCP
+from ocs_ci.ocs.resources.catalog_source import CatalogSource
+from ocs_ci.ocs.resources.csv import CSV
+from ocs_ci.ocs.resources.install_plan import wait_for_install_plan_and_approve
+from ocs_ci.ocs.resources.pod import verify_pods_upgraded
+from ocs_ci.ocs.resources.packagemanifest import (
+    get_selector_for_ocs_operator,
+    PackageManifest,
+)
+from ocs_ci.ocs.resources.storage_cluster import StorageCluster
+from ocs_ci.ocs.utils import setup_ceph_toolbox
+from ocs_ci.utility.utils import (
+    get_latest_ds_olm_tag,
+    get_next_version_available_for_upgrade,
+    get_ocs_version_from_image,
+    load_config_file,
+)
+from ocs_ci.utility.templating import dump_data_to_temp_yaml
+
+log = logging.getLogger(__name__)
+
+
+def get_upgrade_image_info(old_csv_images, new_csv_images):
+    """
+    Log the info about the images which are going to be upgraded.
+
+    Args:
+        old_csv_images (dict): dictionary with the old images from pre upgrade CSV
+        new_csv_images (dict): dictionary with the post upgrade images
+
+    Returns:
+        tuple: with three sets which contain those types of images
+            old_images_for_upgrade - images which should be upgraded
+            new_images_to_upgrade - new images to which upgrade
+            unchanged_images - unchanged images without upgrade
+
+    """
+    old_csv_images = set(old_csv_images.values())
+    new_csv_images = set(new_csv_images.values())
+    old_images_for_upgrade = old_csv_images - new_csv_images
+    log.info(
+        f"Old images which are going to be upgraded: "
+        f"{sorted(old_images_for_upgrade)}"
+    )
+    new_images_to_upgrade = new_csv_images - old_csv_images
+    log.info(
+        f"New images for upgrade: "
+        f"{sorted(new_images_to_upgrade)}"
+    )
+    unchanged_images = old_csv_images.intersection(new_csv_images)
+    log.info(
+        f"Unchanged images after upgrade: "
+        f"{sorted(unchanged_images)}"
+    )
+    return (
+        old_images_for_upgrade,
+        new_images_to_upgrade,
+        unchanged_images,
+    )
+
+
+def verify_image_versions(old_images, upgrade_version):
+    """
+    Verify if all the images of OCS objects got upgraded
+
+    Args:
+        old_images (set): set with old images
+        upgrade_version (packaging.version.Version): version of OCS
+
+    """
+    namespace = config.ENV_DATA['cluster_namespace']
+    number_of_worker_nodes = len(get_typed_nodes())
+    storage_cluster = StorageCluster(
+        resource_name=config.ENV_DATA['storage_cluster_name'],
+        namespace=namespace
+    )
+    osd_count = (
```

Comment:
This was probably missed in some of your rebase but the logic here were changed to use the function `get_osd_count` see here: https://github.com/red-hat-storage/ocs-ci/commit/4132546cd260a80aa85b335299ee8038d0dd228f#diff-40ca86338b56ecd56fb458ca2ea36c3b so please make sure we did not loose any content of that commit here, thanks.

Reply:
fixed